Vagrantfile

```diff
@@ -206,6 +212,67 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       }).strip
     end
 
+    # Then try vsphere
+    config.vm.provider :vsphere do |vsphere, override|
+      # Check if plugin installed
+      # https://github.com/nsidc/vagrant-vsphere
+      if Vagrant.has_plugin?("vagrant-vsphere") then
```

Comment:
You should be able to remove this check.  Unfortunately, it calls this code block even when you are not running with that provider.  The check in verify-prereqs should be sufficient enough.